Quantum-Control-Applications/ESR/T2.py

```diff
@@ -0,0 +1,172 @@
+from qm import SimulationConfig
+from qm.qua import *
+from qm import LoopbackInterface
+from qm.QuantumMachinesManager import QuantumMachinesManager
+from configuration import *
+import matplotlib.pyplot as plt
+from qm.simulate.credentials import create_credentials
+
+###################
+# The QUA program #
+###################
+
+pi_len = 320 // 4
+pi_half_len = 160 // 4
+
+delay_min = 1500 // 4
+delay_max = 20000 // 4
+ddelay = 1000 // 4
+
+delays = np.arange(delay_min, delay_max + 0.1, ddelay)
+
+cooldown_time = int(10e6 // 4)
+
+n_avg = 100
+
+with program() as T2:
+
+    n = declare(int)
+    n_st = declare_stream()
+    delay_len = declare(int)
+    pulse_delay = declare(int)
+    readout_delay = declare(int)
+
+    I = declare(fixed)
+    Q = declare(fixed)
+    I_st = declare_stream()
+    Q_st = declare_stream()
+
+    with for_(n, 0, n < n_avg, n + 1):
+
+        with for_(delay_len, delay_min, delay_len <= delay_max, delay_len + ddelay):
+
+            # wait(cooldown_time)
+
+            reset_phase("ensemble")
+            reset_phase("resonator")
+            reset_frame("ensemble")
+
+            echo = declare_stream(adc_trace=True)
+
+            assign(pulse_delay, delay_len - pi_half_len * 0.5 - pi_len * 0.5)
+            assign(readout_delay, delay_len - pi_len * 0.5 - readout_len * 0.125 - 5)
+
+            wait(8, "switch_1", "switch_2")
+            play("activate", "switch_1", duration=pi_half_len)
+            play("activate", "switch_2", duration=pi_half_len)
+            play("const", "ensemble", duration=pi_half_len)
+
+            wait(pulse_delay, "ensemble", "switch_1", "switch_2")
+
+            frame_rotation_2pi(-0.5, "ensemble")
+            play("activate", "switch_1", duration=pi_len)
+            play("activate", "switch_2", duration=pi_len)
+            play("const", "ensemble", duration=pi_len)
```

Comment:
Same comment as for T1, why do we use this pi/2 - wait - pi - wait sequence? isn't T2 measured with a Ramsey sequence?

Reply:
In a Ramsey sequence in SC qubits you need to apply a return pi/2 to go back to the z-axis to perform a measuremnet. In ESR you perform your measurment in XY plane. In ESR there are three T2, T2* (related to pi/2 and measure), T2_echo (related to pi/2 - variable delay - pi -echo measurement0, and T2_DD (rleate dto T2 measured by XY8 or CPMG)